### ISSUE TITLE
[Linux][Ubuntu 18.04][HDPI] Keybase autostarts with wrong scaling factor

### ISSUE THREAD

**Issue by gianpaolom** [keybase/client#11786]
I am running the latest Keybase on Ubuntu 18.04, GNOME session (Xorg), on a laptop with a res of 3200x1800, and scaling factor of 2. 
After login, Keybase autostarts, but it does not recognise the scaling factor, so the window of Keybase is very small and unreadable.

After some testing I found that there is probably some race condition, in fact if I login into GNOME with autostart disabled, and start Keybase manually after some time after login, it picks up the scaling factor correctly.

So a quick fix for me would have been adding:
```sh
X-GNOME-Autostart-Delay=<N>  # delay time in seconds
```
to `~/.config/autostart/keybase_autostart.desktop`:
```sh
# This file is autogenerated when Keybase starts. Keybase will recreate it if you
# delete it, so to disable autostart you should uncomment the line below.

[Desktop Entry]
Name=Keybase
Comment=Keybase Filesystem Service and GUI
Type=Application
Exec=env KEYBASE_START_UI=hideWindow XDG_CURRENT_DESKTOP=Unity run_keybase
X-GNOME-Autostart-Delay=15  <===========

# Uncomment the following line to disable Keybase autostart:
# Hidden=true
```

However Keybase rewrites that file every time it starts, so nuking my fix.

Would it be possible to allow that parameter to be set in the autostart configuration file?

**Comment by oconnor663** [comment]
I noticed a similar issue back when I was running a desktop just like yours, but I didn't have time to dig into it. I've been hesitant to add a sleep, since it feels like an probably-sometimes-unreliable workaround that doesn't really address the problem. That said, it's probably better than doing nothing.

I'm surprised you find your changes get nuked. As the comment at the bottom of the file describes, we do expect people to be able to edit it and have their edits preserved. This might be a bug...

**Comment by oconnor663** [comment]
Ok the `.autostart` behavior isn't a bug, but it's some clunky behavior that we should probably revamp. The issue is that we sometimes need to change things in that file, and so we pave over it when we don't detect that it's been `Hidden`. I think what we should do instead is 1) arrange for the .desktop file to offload all its interesting work (like setting `KEYBASE_START_UI=hideWindow`) to some wrapper script we control, and 2) make a note in `~/.config/keybase` when we create it, so that we never touch it again afterwards. I'll file a ticket for this.

**Comment by gianpaolom** [comment]
To be complete I found it out as I noticed that when I login and the X session starts loading, I can see the mouse pointer not scaled yet. Then all the rest of the desktop loads and it's scaled 2.
So I reckon the race condition happens there: the scaling is not applied immediately and if keybase is quicker it detects the full resolution instead of the scaled one.

@oconnor663 so thanks for your answer and for addressing the issue
